docs/4.1/architecture/nodes.md

```diff
@@ -0,0 +1,109 @@
+`sshd`.
+
+![Node Service ping API](../img/node_service_api.svg)
+
+All cluster Nodes keep the Auth Server updated with their status with periodic
+ping messages. They report their IP addresses and values of their assigned
+labels. Nodes can access the list of all Nodes in their cluster via the
+[Auth Server API](./auth/#auth-api).
+
+!!! tip "Tip"
+    In lightweight environments, such as IoT deployments, it is possible to entirely replace `sshd` with the `node` service <!--other examples?-->.
+
+The `node` service provides SSH access to every node with all of the following clients:
+
+* [OpenSSH: `ssh`](../admin-guide/#using-teleport-with-openssh)
+* [Teleport CLI client: `tsh ssh`](../cli-docs/#tsh-ssh)
+* [Teleport Proxy UI](./proxy/#web-to-ssh-proxy) accessed via a web browser.
+
+Each client is authenticated via the [Auth Service](./auth/#authentication-in-teleport) before being granted access to a Node.
+
+## Node Identity on a Cluster
+
+Node Identity is defined on the Cluster level by the certificate a node possesses.
+
+![Node Identity](../img/node_identity.svg)
+
+This certificate contains information about the node including:
+
+* The **host ID**, a generated UUID unique to a node
+* A **nodename**, which defaults to `hostname` of the node, but can be configured.
+* The **cluster_name**, which defaults to the `hostname` of the auth server, but can be configured
+* The node **role** (i.e. `node,proxy`) encoded as a certificate extension
+* The cert **TTL** (time-to-live)
+
+A Teleport Cluster is a set of one or more machines whose public keys are signed by the same certificate authority (CA) operating in the Auth Server. A certificate is issued to a node when it joins the cluster for the first time. Learn more about this process in the [Auth Guide](./auth/#authentication-in-teleport).
+
+!!! warning "Single-Node Clusters are Clusters"
+    Once a Node gets a signed certificate from the Node CA, the Node is considered a member of the cluster, even if that cluster has only one node.
+
+## Connecting to Nodes
+
+When a client requests access to a Node, authentication is always performed through a cluster proxy. When the proxy server receives a connection request from a client it validates the client's credentials with the Auth Service. Once the client is authenticated the proxy attempts to connect the client to the requested Node.
+
+There is a detailed walk-through of the steps needed to initiate a connection to a node in the [Architecture Overview](./architecture).
+
+![Proxy Connection between client and node](../img/proxy_client_connect.svg)
+
+[Session state](./auth/#auth-state) is stored on the Auth Server rather than on the Node. Each node is completely stateless and holds no secrets such as keys or passwords.
+
+## Cluster State
+
+Cluster state is stored in a central storage location configured by the Auth Server. This means that the Node Service is completely stateless.
```

Comment:
We're asseting that the node service is stateless twice in two lines - might be best to switch it up a little.